core/src/main/java/org/elasticsearch/search/aggregations/ParsedAggregation.java

```diff
@@ -63,11 +66,12 @@ public final String getName() {
 
     //TODO the only way to avoid duplicating this method is making Aggregation extend ToXContent
     //and declare toXContent as a default method in it. Doesn't sound like the right thing to do.
+    @Override
     public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
         //TODO move TYPED_KEYS_DELIMITER constant out of InternalAggregation
         // Concatenates the type and the name of the aggregation (ex: top_hits#foo)
         builder.startObject(String.join(InternalAggregation.TYPED_KEYS_DELIMITER, getType(), name));
-        if (metadata.isEmpty() == false) {
+        if (this.metadata != null) {
```

Comment:
I changed this (and the way the class interacts with the internal field) because I wanted to get close to what InternalAggregation allows for metadata. I think it can be null there, in which case the method returns null and we don't write anything to xContent.